### ISSUE TITLE
What do we need in secrets.json file ?

### ISSUE THREAD

**Issue by adri1mart1** [jannisborn/coinbot#36]
As stated in the README, this part is not so straighforward.

Based on the code I've seen, I've started to create a json file like so:

```json
{
    "telegram-token": "xx",
    "anyscale": "xx",
    "file_link": "xx",
    "slack": "xx"
}
```

An easy solution would be to create a secrets.json.template file with the expected structure. 
In addition, it would be great to have a small explanation in the README on each field to indicate what's expected, at the moment, I only know how to get the telegram-token.

Could you clarify what should I use to make it work please ?

I'm looking forward to try this bot :)

**Comment by jannisborn** [comment]
Hi,
Thanks for your interest in the project :) Can you explain a bit your motivation for using this code? Do you want to build a similar app etc? If you simply want to use the bot, you can already do so by texting to [@coincollectionbot](https://t.me/coincollectionbot) or by scanning the QR code from the README. The bot is up and running, I use it personally to collect more unique euro coins :)

You will only need the tokens if you want to set a similar bot yourself.
- `anyscale`: This token is to use a LLM for translating the user's message or extract certain data. Note that this token will be no longer needed because Anyscale shutted down its multi-tenant LLM API as of August 1. I refactored the code to use Together instead (together.ai), see open PR #37 which will be merged soon. Hence, in the future a `together` token will be needed which can be obtained from their website after creating an account
- `telegram-token` is a token for your own new telegram bot
- `file_link` is a link to a downloadable database file. In my case, it's an `xlsm` file which contains all coins I have already collected
- `slack` token is used to send messages to a slack channel. I use this to post to my own slack channel so that I can see if my friends collect new coins via the bot

Hope this helps, happy to tell more and happy coin collecting

<img width="1060" alt="image" src="https://github.com/user-attachments/assets/850c8ed8-0b7f-455b-9765-472c29c7a916">

**Comment by adri1mart1** [comment]
Thank you for your reply. 

I started a collection many years ago and over the year I never used a tool or an app to follow my coins. I tried some but most of it I was disappointed. Use a website is cool but if the website shutdown, that's very bad. Same for an app, etc.. 

I had the idea of a bot telegram looking at the telegram API few years ago but I did not have the courage to really start because you have news coins very regularly, once you start a project like that, you need to have your database up with new coins.

I'm really happy that I found about your project few days ago, you did a really good job (way better than I could have done). I would have never though about integrating an AI to interact with the user (not really my domain of expertise) but that's very smart !

Basically I would love to use this app and maybe I'll use the one online. 

However, I'm also very careful with my data and I would love to have my data on my personal server. I believe if I use [@coincollectionbot](https://t.me/coincollectionbot), It will create an xlsm file of my collection somewhere on your server ? 

I also don't know yet if slack is optional ? I don't use slack personally. From what I understood, It's simply for you to maybe help you collect more coins or the idea it to share any findings by anyone so that you can be notified, discuss and maybe exchange with the community about a specific coin ?

I don't want to modify you bot right now, just want to use it first. However, I would love to own my data.

**Comment by jannisborn** [comment]
Hi,

Alright that sounds great. To help you further and say whether this codebase can be really useful for you, I need to understand a bit better what you want your bot to do. The purpose of this bot for me is NOT to _create_ my own collection but to _access_ it. Basically, the `xlsm` file is an excel sheet that contains my collection. 

It looks like this:
<img width="634" alt="image" src="https://github.com/user-attachments/assets/5b5f0d83-5441-438e-a674-867827b6de13">

Green means collected, white means missing and black mean that this coin was not printed in that year. The value just contains the number of mints. When I collect a coin, I dont want to manually check in the spreadsheet whehter I already have the coin. I just text the bot. The bot also shows me tons of statistics about my collection. You can also collect new coins with the bot. But the bot NEVER writes back to your `xlsm` file, that is not safe to do. Therefore, I use a concept that I call "staging". Staging the coins means it is not added to the collection (in the `xlsm` file) but it has been registered by the bot and I am (manually) adding it soon. 

I assume you are collecting euro coins? Otherwise it will be hard to transfer. 

If you want to use the bot with your own collection, you can do it but:
- you will need to create a `xlsm` file that follows the same structure as my collection file. That is not a problem, I can share my file as an empty template (basically all cells are white). 
- this file will need to be available via a public link. In my case it is a Dropbox link. The link is public but I am the only one who has it, so I consider my collection safe.
- you need access to a server where the bot is running. On the server, clone this repo, install the environment, add your secret file and then start the bot as described in the README

Finally, yes, slack is optional. Currently the code will break without a slack token, but in the next PR I will fix this so that no slack is used if there is no `slack` key in the secrets file.

Let me know if you need more help

**Comment by adri1mart1** [comment]
That's very much clearer, thanks.

I would love to get the xlsm template and try to run the bot to start monitor my own collection as well then. 
I've tried today a few interactions with your both and that seems interesting. 

Few additional questions:
- Any plans or future ideas of features you would love to see with this project ?
- What strategy do you have to stay updated with new coins coming out ? Adding all new coins manually seems that it takes some time, I see you have to scrapping scripts but I did not try to run them.

Regards,
Adrien M.

**Comment by jannisborn** [comment]
Cool, I can send you the template via email. I cant attach it here since `xlsm` is not supported. A few notes about the file:

- It has different sheets, one for German coins (because those are by the far the most because of the different minting locations), all other EU countries, and the special coins
- Black means coin was not minted at all or less than 1M times (I consider < 1M so rare that it is practically impossible to collect it just from circulation). White means coin was not collected. Green means coin was collected
- You MUST use the same green tone that is used on the right side of the sheet to compute the statistics. Actually, there is a macro running (this is why the file is `xlsm` and not `xlsx`) which identifies cells with the same color and thus computes the statistics shown on the right
- It should be straightforward to extend the template in the next years to new coins

Regarfing your questions:
- I consider the bot pretty much complete. Keep an 👁️ on the open issues here, I use this to track what I plan to integrate. The concept of "staging" coins which I described above is the last one I added, and it's still in development, I will create a PR soon
- Good question, currently I do it manually. I use the script once per year to identify the new special coins. For all the regular coins, I'm checking online which were minted and how often. This, honestly, is very tricky to find. If you know any resource for this I would REALLY appreciate! The only one I know is this: https://www.2euro-overview.eu/auflagen.php?land=it But I think it is maintained by a fan so not the most safe/accurate information

Let me know if you need more help with the setup!!

**Comment by adri1mart1** [comment]
Great thank, you can send it to pub at martegy dot net
I'll try that in the coming days and let you know about it. 
Regards,

**Comment by jannisborn** [comment]
Closing here but let me know if you need any help
